auth/team_token.go

```diff
@@ -8,17 +8,32 @@ import (
 	"context"
 	"crypto"
 	"fmt"
+	"strings"
 	"time"
 
 	"github.com/pkg/errors"
 	"github.com/tsuru/tsuru/permission"
 	"github.com/tsuru/tsuru/servicemanager"
 	"github.com/tsuru/tsuru/storage"
+	appTypes "github.com/tsuru/tsuru/types/app"
 	authTypes "github.com/tsuru/tsuru/types/auth"
 	permTypes "github.com/tsuru/tsuru/types/permission"
+	"github.com/tsuru/tsuru/types/quota"
 	"github.com/tsuru/tsuru/validation"
 )
 
+// TsuruTokenEmailDomain is the e-mail domain used to fake users from a team
+// token.
+//
+// NOTE: Following the RFC2606, the top-level domain ".invalid" is reserved,
+// so regular Tsuru users should not be able to register using it. See more:
+// https://www.rfc-editor.org/rfc/rfc2606
+const TsuruTokenEmailDomain = "token.tsuru.invalid"
+
+func IsEmailFromTeamToken(email string) bool {
+	return strings.HasSuffix(email, fmt.Sprintf("@%s", TsuruTokenEmailDomain))
```

Comment:
is it possible to arrive at this line with a url-encoded email? 
eg: `example%40token.tsuru.invalid`.
This would return false, although it should return true

Reply:
I don't think so :crossed_fingers: